gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -380,11 +385,23 @@ public void remove(Spec spec, Properties headers) throws IOException {
     // .. this will work for Identity compiler but not always for multi-hop.
     // Note: Current logic assumes compilation is consistent between all executions
     if (spec instanceof FlowSpec) {
+      URI specUri = spec.getUri();
       //Send the dag to the DagManager to stop it.
       //Also send it to the SpecProducer to do any cleanup tasks on SpecExecutor.
       if (this.dagManager.isPresent()) {
-        _log.info("Forwarding cancel request for flow URI {} to DagManager.", spec.getUri());
-        this.dagManager.get().stopDag(spec.getUri());
+        _log.info("Forwarding cancel request for flow URI {} to DagManager.", specUri);
+        this.dagManager.get().stopDag(specUri);
+      }
+      if (this.dagProcessingEngine.isPresent()) {
+        String flowGroup = FlowSpec.Utils.getFlowGroup(specUri);
+        String flowName = FlowSpec.Utils.getFlowName(specUri);
+        List<Long> flowExecutionIds = this.newDagManager.get().getJobStatusRetriever().getLatestExecutionIdsForFlow
+            (flowName, flowGroup, 10);
+        _log.info("Found {} flows to cancel.", flowExecutionIds.size());
+        for (long flowExecutionId : flowExecutionIds) {
+          DagActionStore.DagAction dagAction = DagManagerUtils.createDagAction(flowGroup, flowName, String.valueOf(flowExecutionId), DagActionStore.FlowActionType.KILL);
+          this.dagProcessingEngine.get().addDagAction(dagAction);
```

Comment:
At this point we should also/instead add a line to store the new dag action in dagAction store so we can receive this in changeMonitor across all hosts. Remember we want all actions persisted in MySQL and then coordination across hosts.